### PR TITLE
refactor coscheduling to use controller-runtime client

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

As #485 stated, we're migrating all plugins and controllers to use controller-runtime client. With the migration's completion, we should be able to eliminate all typed clients along with its generated bits.

This PR aims to migrate coscheduling.

#### Which issue(s) this PR fixes:

Part of #485

#### Special notes for your reviewer:

This PR also polished existing tests.

#### Does this PR introduce a user-facing change?

```release-note
migrate coscheduling to use controller-runtime client 
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
cc @denkensk @zwpaper for review.

**Comment by PiotrProkop** [comment]
LGTM will wait for @denkensk or @zwpaper to add proper lgtm label.

File: pkg/coscheduling/core/core.go [hunk 1]
```diff
@@ -236,27 +230,17 @@ func (pgMgr *PodGroupManager) DeletePermittedPodGroup(pgFullName string) {
...
 // GetPodGroup returns the PodGroup that a Pod belongs to in cache.
 func (pgMgr *PodGroupManager) GetPodGroup(pod *corev1.Pod) (string, *v1alpha1.PodGroup) {
 	pgName := util.GetPodGroupLabel(pod)
 	if len(pgName) == 0 {
 		return "", nil
 	}
-	pg, err := pgMgr.pgLister.PodGroups(pod.Namespace).Get(pgName)
-	if err != nil {
+	var pg v1alpha1.PodGroup
+	if err := pgMgr.client.Get(context.Background(), types.NamespacedName{Namespace: pod.Namespace, Name: pgName}, &pg); err != nil {
```
Comment (kerthcet) [inline pkg/coscheduling/core/core.go:240]: nit: can we unified to use context.TODO?

**Review by zwpaper — COMMENTED** [review]
it also LGTM, only have a question

File: pkg/coscheduling/core/core.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Huang-Wei) [reply pkg/coscheduling/core/core.go:240]: it's basically the same.

[the remark this answers is not in the document]
Comment (Huang-Wei) [reply pkg/coscheduling/core/core.go:240]: updated to reuse context.

**Comment by Huang-Wei** [comment]
@zwpaper @kerthcet I refactored a bit to pass in context in places where context obj is derivable. However for some places like Less(), it's still infeasible w/o changes in scheduler framework for now https://github.com/kubernetes-sigs/scheduler-plugins/pull/652#discussion_r1376600818

**Comment by kerthcet** [comment]
LGTM.

**Pull request merged**